one-ui/src/main/java/com/marklogic/hub/oneui/managers/SavedQueriesManager.java

```diff
@@ -0,0 +1,43 @@
+package com.marklogic.hub.oneui.managers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.MarkLogicServerException;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.dataservices.SavedQueriesService;
+import com.marklogic.hub.oneui.exceptions.BadRequestException;
+import com.marklogic.hub.oneui.exceptions.DataHubException;
+
+public class SavedQueriesManager {
+
+    private DatabaseClient finalDataServiceClient;
+
+    public SavedQueriesManager(HubConfig hubConfig) {
+        finalDataServiceClient = hubConfig.newFinalClient();
+    }
+
+    public JsonNode saveOrUpdateQueryDocument(JsonNode queryDocument) {
+        if (queryDocument == null || queryDocument.get("savedQuery") == null) {
```

Comment:
I think all of this logic should be in the DS endpoint - this is all validation logic, and similar to artifacts, we want to enforce this in the DS, not in Java code.

By doing that, you don't even need this class - the controller can just interact directly with the SavedQueryService. 

Also, @ryanjdew - this seems like a case in the future where we could use JSON schema to validate the incoming JSON so we don't have to write a lot of imperative code like this, right?